Mage/Archivist/Archivist.cs

```diff
@@ -132,25 +130,39 @@ public class Archivist : EventEmitter<VaultValue> {
 
 	// Remove a vault value from the cache by it's cache key name
 	public void DeleteCacheItem(string cacheKeyName) {
-		if (!_cache.ContainsKey(cacheKeyName)) {
-			return;
-		}
+		lock ((object)this) {
+			logger.debug("Deleteing cache item: " + cacheKeyName);
```

Comment:
yes there is, we haven't not starting the delete op until the lock is acquired.
I do not want to get logging when the op hasn't even begun yet. It would confuse the developer terribly